reconcile/external_resources/manager.py

```diff
@@ -179,21 +180,22 @@ def _get_desired_objects_reconciliations(self) -> set[Reconciliation]:
         return r
 
     def _get_deleted_objects_reconciliations(self) -> set[Reconciliation]:
-        desired_keys = set(self.er_inventory.keys())
-        state_resource_keys = self.state_mgr.get_all_resource_keys()
-        deleted_keys = state_resource_keys - desired_keys
-        r: set[Reconciliation] = set()
-        for key in deleted_keys:
+        to_reconcile: set[Reconciliation] = set()
+        for key in [k for k, v in self.er_inventory.items() if v.marked_to_delete]:
```

Comment:
multiple conditions `for` and `if` in one statement, worth extract a var like before 

```python
deleted_keys = (k for k, v in self.er_inventory.items() if v.marked_to_delete)
for key in deleted_keys:
    ...
```

And please doc if the item is removed instead of mark to delete, it will be orphan
```

Reply:
`multiple conditions for and if in one statement, worth extract a var like before`  I think we should define guidelines on these things because sometimes I get the contrary message (avoid defining a variable to just use it one time). 

Yup, abandoning resources will be documented.

Reply:
it depends on context, extract variable and inline variable are common refactoring pair, if it's hard to see the intention of a expression, then extract, if it's already clear from context like method name, then inline:

some examples:
* https://refactoring.guru/extract-variable
* https://refactoring.guru/inline-temp